### PR TITLE
Allow uploading multiple files

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

- Fix styling of the upload button
- Allow uploading multiple files
- Show upload progress on button
- Highlight add media button when no local media found

https://user-images.githubusercontent.com/1444314/153923732-f028c28b-661e-43fb-bbd8-dd012ff3ba45.mp4

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by matthiasdebaat** [comment]
Really nice! When uploading, can we make the icon into an animated spinner?

**Comment by balloob** [comment]
done.

https://user-images.githubusercontent.com/1444314/153927998-c6ba58dc-a930-4cfc-b82d-cfccd30588cf.mp4

File: src/panels/media-browser/ha-panel-media-browser.ts [hunk 1]
```diff
@@ -240,36 +265,57 @@ class PanelMediaBrowser extends LitElement {
...
-            {
-              reason: err.message || err,
+    input.multiple = true;
+    input.addEventListener(
+      "change",
+      async () => {
+        const files = input.files!;
+        const target = this._currentItem!.media_content_id!;
+
+        try {
```
Comment (bramkragten) [inline src/panels/media-browser/ha-panel-media-browser.ts:281]: What will this try catch, as the try inside the loop will catch everything?

Comment (balloob) [reply src/panels/media-browser/ha-panel-media-browser.ts:281]: I use it for the `finally`.

Comment (bramkragten) [reply src/panels/media-browser/ha-panel-media-browser.ts:281]: The finally isn't needed then right? If it will not throw anyway?

Comment (balloob) [reply src/panels/media-browser/ha-panel-media-browser.ts:281]: if an error happens in the inner try, we do a `return`.

So now in both error case + when finished will we set uploading to 0 and refresh.

Comment (bramkragten) [reply src/panels/media-browser/ha-panel-media-browser.ts:281]: Why `return` and not `break`?

Comment (balloob) [reply src/panels/media-browser/ha-panel-media-browser.ts:281]: I don't know 😄

Comment (balloob) [reply src/panels/media-browser/ha-panel-media-browser.ts]: Fixed

**Review by zsarnett — APPROVED** [review]

**Pull request merged**